### PR TITLE
Fix lifetime option

### PR DESCRIPTION
**Pull request by Helen-Mobify**
Fix lifetime option

Status: **Opened for visibility**

Reviewers: @tedtate @donnielrt @jansepar 
JIRA: https://github.com/mobify/split-test/issues/7

## Changes
- Fix the lifetime option where defining the option does not change the cookie's expiry time
- add test to validate the lifetime option

### CODE REVIEW

File: dist/split-test.js [hunk 1]
```diff
@@ -25,6 +25,7 @@
         }
 
         this.cookieDomain = options.cookieDomain || window.location.hostname;
+        this.lifetime = options.lifetime || 0;
```
Comment (jansepar) [inline dist/split-test.js]: Is 0 the default we want here? Won't this cause the cookies to not persist?

Comment (jansepar) [inline dist/split-test.js]: Ah, it won't because if it's not greater then 0, we default to 30 seconds. Still though, you have to look deep in the code for that. I feel like it would be better to set this to the 30 day default right here so that it's less confusing for developers who look at this code later.

**Comment by jansepar** [comment]
@Helen-Mobify yesss thank you for opening up this PR!

**Comment by Helen-Mobify** [comment]
@jansepar I am not sure which is a better approach:
Define the default lifetime (30 days) in the SplitTest constructor
OR
set a default expiry in the setCookie function if no lifetime is defined
OR
both?

File: dist/split-test.js (hunk 1 above)
Comment (Helen-Mobify) [inline dist/split-test.js]: Yea. I wasn't quite sure if this is the best approach.

**Comment by jansepar** [comment]
Both - setCookie is still a public method, so it should still guard against edge cases. I think it makes sense for `setCookie` and the constructor to have separate defaults, even if they actually are the same.

**Comment by jansepar** [comment]
Mostly because I've seen people take *just* the setCookie method from the library and use nothing else :)

**Comment by Helen-Mobify** [comment]
Cool. I'll make that change.

**Comment by Helen-Mobify** [comment]
@jansepar Eyes please

File: dist/split-test.js [hunk 2]
```diff
@@ -17,6 +17,8 @@
     }
 }(this, function() {
 
+    var DEFAULT_LIFETIME = 30 * 24 * 3600;
```
Comment (donnielrt) [inline dist/split-test.js]: If we're only using this in setCookie, where we're multiplying by 1000, can't we just set it here? i.e.:

 var DEFAULT_LIFETIME = 30 * 24 * 60 * 60 * 1000; // In ms

Comment (Helen-Mobify) [inline dist/split-test.js]: This variable is also used in the SplitTest constructor. So if we want to change it, we need to make sure that all time variable is in the same unit (ie. all in seconds or milliseconds)

Comment (donnielrt) [inline dist/split-test.js]: Yup, IMO we should both to ms, since that's the actual desired unit. Also worth adding a 
comment to the splitTest function describing expected parameters and units, since Shawn mentioned people copy this into other places.

Comment (jansepar) [inline dist/split-test.js]: +1 to the way Don made the calculation - it makes it a bit more clear how to tell how long things are by breaking them down into units.

Comment (Helen-Mobify) [inline dist/split-test.js]: cool. milliseconeds it is.

**Comment by Helen-Mobify** [comment]
@donnielrt @jansepar Anything else? or good to merge?

**Comment by donnielrt** [comment]
@Helen-Mobify tests fail ;) we should integrate CircleCI into this repo to catch test fails.

**Comment by Helen-Mobify** [comment]
@jansepar is this consider as breaking change? 
The version before is using seconds and this version will now be using milliseconds.

**Comment by Helen-Mobify** [comment]
@donnielrt ahh right forgot to update test

**Comment by jansepar** [comment]
Ah true the setCookie function used to take seconds... I wouldn't want to change that because if they get the latest version of split-test, it would have unexpected behavior. Let's make sure not to break backwards compatibility.

We could just make it so that setCookie still takes seconds, but the constructor takes milliseconds, then we can just convert it.

**Comment by Helen-Mobify** [comment]
alight

**Comment by Helen-Mobify** [comment]
@donnielrt I can't add this repo into circleci

**Comment by donnielrt** [comment]
Yeah, +1 to keeping it in seconds for setCookie, because even if we bumped up the version number, people might easily miss that change.

**Comment by jansepar** [comment]
I'm heading to eye surgery so feel free to +1 amongst yourselves :)

—

Shawn Jansepar, Lead Software Engineer

  Better Mobile Shopping.

mobify.com <http://www.mobify.com/> | @shawnjan8

Get deep insights into mobile shopping strategy on Mobify Insights
<http://insights.mobify.com/>.

**Comment by donnielrt** [comment]
:+1: from me.

**Comment by Helen-Mobify** [comment]
:boom: merged!
